script.js

```diff
@@ -1,4 +1,275 @@
+    return `Both of your numbers are ${player1ConcatenatedNumber}. <br><br>It's a tie! <br><br>${howToReset}`;
+  }
+
+  //Check if player 1 won
+  if (player1ConcatenatedNumber > player2ConcatenatedNumber) {
+    return `Player 1's number is ${player1ConcatenatedNumber}. <br>Player 2's number is ${player2ConcatenatedNumber}. <br><br>Player 1 won! <br><br>${howToReset}`;
+  }
+
+  //Check if player 2 won
+  if (player2ConcatenatedNumber > player1ConcatenatedNumber) {
+    return `Player 1's number is ${player1ConcatenatedNumber}. <br>Player 2's number is ${player2ConcatenatedNumber}. <br><br>Player 2 won! <br><br>${howToReset}`;
+  }
+};
+
+//Create a function that resets the game
+var gameReset = function () {
+  hasPlayer1Rolled = 0;
+  hasPlayer2Rolled = 0;
+  player1DiceRolls = [];
+  player2DiceRolls = [];
+  player1ConcatenatedNumber = 0;
+  player2ConcatenatedNumber = 0;
+  timeToReset = 0;
+  return `The game has been reset! You may play again by pressing the "Submit" button.`;
+};
+
+//Create a function that receives the user input and returns the correct output statements
+var beatThatGame = function (userInput) {
+  //Make the input entirely lowercase
+  userInput = userInput.toLowerCase();
+
+  if (
+    !(userInput == "dice 1" || userInput == "dice 2") &&
+    ((hasPlayer1Rolled == 1 && player1ConcatenatedNumber == 0) ||
+      (hasPlayer1Rolled == 1 &&
+        hasPlayer2Rolled &&
+        player2ConcatenatedNumber == 0))
+  ) {
+    return;
+  }
+  if (userInput == "" && !(hasPlayer1Rolled == 1 && hasPlayer2Rolled == 1)) {
+    //Check if there is no input, which means the players are rolling the dice. Call the dice rolling functions
+    rollTheDice();
+    return statementForDiceRolled();
+  }
+
+  //Check if input is "dice 1" or "dice 2", which means the players are choosing which dice is the first number and which is the second. Call the concatenating functions
+  if (userInput == "dice 1" || userInput == "dice 2") {
+    concatenator(userInput);
+    return statementForConcatenation(userInput);
+  }
+
+  //Check if input is "compare", which means its time to compare the two numbers and see who won
+  if (
+    userInput == "compare" &&
+    hasPlayer1Rolled == 1 &&
+    hasPlayer2Rolled == 1
+  ) {
+    timeToReset = 1;
+    return comparingWinnerStatement();
+  }
+
+  //Check if input is "reset", which means that it is time to reset the game
+  if (userInput == "reset" && timeToReset == 1) {
+    return gameReset();
+  }
+  console.log("Player 1: ", player1DiceRolls, "Player 2: ", player2DiceRolls);
+};
+
+//Create a function that takes the user input and outputs the correct instructions if the user types invalid input
+var invalidInputChecker = function () {
+  //Check if both players have not rolled the dice yet
+  if (hasPlayer1Rolled == 0 && hasPlayer2Rolled == 0) {
+    return `Please press the "submit" button without typing anything in the input box to start the game for player 1`;
```

Comment:
Do we really need this specific condition where the input box needs to be empty in order to start the game? Seems oddly specific, why not just ignore whatever input and start the game regardless once the submit button is clicked, if both players have not rolled any die

To build on this further, would be a good idea to output some message to the user immediately on page load, instructing them how to start the game. This can be done by changing your HTML, do explore and let me know if you have questions. Currently, when I load up your game, I am greeted with an empty text input and empty output box, and it is not immediately clear what I have to do. Another point related to user experience to consider